File: app/src/main/java/com/annushkaproject/programmerscalculator/model/ProgrammerCalcModel.java
package com.annushkaproject.programmerscalculator.model;
import com.annushkaproject.programmerscalculator.model.CalculationModel;
import java.math.BigDecimal;
import static com.annushkaproject.programmerscalculator.model.WordLength.QWORD;
public class ProgrammerCalcModel extends CalculationModel {
    private WordLength wordLength;
    public WordLength getWordLength() { return wordLength; } 
    public void setWordLength(WordLength wordLength) { this.wordLength = wordLength; } 
    public ProgrammerCalcModel() { this.wordLength = QWORD; } 
    public ProgrammerCalcModel(BigDecimal firstValue, Operator operator, WordLength wordLength) {
        super(firstValue, operator);
        this.wordLength = wordLength;
    }

    public ProgrammerCalcModel(BigDecimal firstValue, BigDecimal secondValue, Operator operator, WordLength wordLength) {
        super(firstValue, secondValue, operator);
        this.wordLength = wordLength;
    }
    public void updateValues(String text, Mode mode) {
        if (this.getOperator() == null) {
            setFirstValue(Long.parseLong(text, mode.getBase()));
        } else {
            setSecondValue(Long.parseLong(text, mode.getBase()));
        }
    }

}

File: app/src/main/java/com/annushkaproject/programmerscalculator/model/ProgrammerCalcModel.kt
package com.annushkaproject.programmerscalculator.model

import java.math.BigDecimal

class ProgrammerCalcModel : CalculationModel {
    var bytelengthenum: int_size_enum

    constructor() {
        bytelengthenum = int_size_enum.l8
    }

    constructor(firstValue: BigDecimal?, operator: Operator?, bytelengthenum: int_size_enum) : super(firstValue, operator) {
        this.bytelengthenum = bytelengthenum
    }

    constructor(firstValue: BigDecimal?, secondValue: BigDecimal?, operator: Operator?, bytelengthenum: int_size_enum) : super(firstValue, secondValue, operator) {
        this.bytelengthenum = bytelengthenum
    }

    fun updateValues(text: String, modeenum: mode_enum) {
        if (operator == null) setFirstValue(text.toLong(modeenum.base).toDouble()) else setSecondValue(text.toLong(modeenum.base).toDouble())
    }
}